### PR TITLE
Unignore ihm elb input

### PR DESCRIPTION
**Pull request by zancas**
This PR removes `dbg` and `ignore` statements from the code.

Once it lands several more tests will run!

One outstanding TODO, is to upgrade tests that do _not_ include assert statements.

### CODE REVIEW

File: src/utils/test.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ Result:
      }, ...
   ],
   "upgrades": {                (object) status of network upgrades
-     "xxxx" : {                (string) branch ID of the upgrade
+     "xxxx": {                (string) branch ID of the upgrade
```
Comment (dannasessha) [inline src/utils/test.rs:42]: Okay, looks like I was embarassingly wrong in our chat: this space **does** exist in the help output.

Therefore, I would like to retain this space for the time being.

I think this is a case for the scrubber to address, though I'm surprised that we didn't notice an error here before, if this vector was used in any passing test. We'll have to track down what happened.

File: src/lib.rs [hunk 2]
```diff
@@ -56,7 +56,7 @@ pub fn check_success(output: &std::process::ExitStatus) {
 }
 
 pub fn interpret_help_message(raw_command_help: &str) -> serde_json::Value {
-    let (cmd_name, result_data) = extract_name_and_result(raw_command_help);
+    let (_cmd_name, result_data) = extract_name_and_result(raw_command_help);
```
[the remark this answers is not in the document]
Comment (zancas) [reply src/lib.rs:59]: The warning needs suppression, yes.

File: src/lib.rs [hunk 3]
```diff
@@ -118,32 +118,25 @@ fn annotate_object(result_chars: &mut std::str::Chars) -> serde_json::Value {
     loop {
         match result_chars.next().unwrap() {
             '}' => {
-                //dbg!("end brace");
-                //dbg!(&viewed);
                 if viewed.trim().is_empty() {
                     break;
                 }
                 if viewed.trim() == ", ..." {
-                    dbg!("trailing dots in Object");
```
[the remark this answers is not in the document]
Comment (zancas) [reply src/lib.rs:127]: This should be handled in scrubbing, so why keep the match?

File: src/lib.rs [hunk 4]
```diff
@@ -655,77 +631,101 @@ mod unit {
...
             interpret_help_message(test::EXTRABRACKETS2_HELP_GETINFO);
         assert_eq!(valid_help_in, test::valid_getinfo_annotation());
     }
 
-    #[ignore]
     #[test]
     fn interpret_help_message_getblockchaininfo_softforks_fragment() {
         let expected_incoming = test::GETBLOCKCHAININFO_SOFTFORK_FRAGMENT;
-        let expected_results = r#"{"softforks":"[{\"enforce\":\"{\\\"found\\\":\\\"Decimal\\\",\\\"required\\\":\\\"Decimal\\\",\\\"status\\\":\\\"bool\\\",\\\"window\\\":\\\"Decimal\\\"},\",\"id\":\"String\",\"reject\":\"{\\\"found\\\":\\\"Decimal\\\",\\\"required\\\":\\\"Decimal\\\",\\\"status\\\":\\\"bool\\\",\\\"window\\\":\\\"Decimal\\\"}\",\"version\":\"Decimal\"}],"}"#;
+        let expected_result = serde_json::json!({"softforks":[{"enforce":{"found":"Decimal","required":"Decimal","status":"bool","window":"Decimal"},"id":"String","reject":{"found":"Decimal","required":"Decimal","status":"bool","window":"Decimal"},"version":"Decimal"}]});
```
[the remark this answers is not in the document]
Comment (zancas) [reply src/lib.rs:671]: Yes.   Testing a subcomponent still gives you information you wouldn't otherwise have.

File: src/utils/test.rs (hunk 1 above)
Comment (zancas) [reply src/utils/test.rs:42]: Why do you believe that?  I thought I looked at it in the Copy Machine, and it didn't.

**Comment by zancas** [comment]
Alright, I force-pushed this updated PR now that @AloeareV 's mutability removal is landed.

**Comment by zancas** [comment]
OK, this PR is not ready to land yet.

File: src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (zancas) [reply src/lib.rs:127]: rm'ed

File: src/lib.rs [hunk 5]
```diff
@@ -655,77 +631,101 @@ mod unit {
         dbg!(interpret_help_message(test::UPGRADES_IN_OBJ_EXTRACTED));
     }
 
-    // ----------------interpret_help_message : ignored---------------
+    // ----------------interpret_help_message---------------
```
[the remark this answers is not in the document]
Comment (zancas) [reply src/lib.rs:634]: rmd

File: src/lib.rs [hunk 6]
```diff
@@ -655,77 +631,101 @@ mod unit {
...
-    #[ignore]
     #[test]
     fn interpret_help_message_getblockchaininfo_enforce_and_reject_fragment() {
         let expected_incoming =
             test::GETBLOCKCHAININFO_ENFORCE_AND_REJECT_FRAGMENT;
-        let expected_results = r#"{"enforce":"{\"found\":\"Decimal\",\"required\":\"Decimal\",\"status\":\"bool\",\"window\":\"Decimal\"},","id":"String","reject":"{\"found\":\"Decimal\",\"required\":\"Decimal\",\"status\":\"bool\",\"window\":\"Decimal\"}","version":"Decimal"}"#;
-        let interpreted =
-            format!("{}", interpret_help_message(expected_incoming));
+        let expected_results = serde_json::json!({"enforce":{"found":"Decimal",
+                                                             "required":"Decimal",
```
[the remark this answers is not in the document]
Comment (zancas) [reply src/lib.rs:685]: You're welcome to try.  We need to standardize formatting, but that's a separate concern.

File: src/utils/test.rs (hunk 1 above)
Comment (zancas) [reply src/utils/test.rs]: OK, I modified our core logic such that we handled this inconsistency.  If you want to add logic that establishes a standard that the help docs have to conform to, then please do so.   In the meantime, we need this code to work.

**Comment by zancas** [comment]
OK, it is now ready AFAIK.

File: src/lib.rs [hunk 7]
```diff
@@ -532,7 +509,7 @@ mod unit {
         let mut special_nested_blockchaininfo =
             &mut test::SPECIAL_NESTED_GETBLOCKCHAININFO.chars();
         let annotated = annotate_result(&mut special_nested_blockchaininfo);
-        let expected_result = serde_json::json!({"xxxx":{"name":"String"}});
+        let expected_result = serde_json::json!({"xxxx" :{"name":"String"}});
```
Comment (dannasessha) [inline src/lib.rs:512]: I don't believe this space should make any difference in the context of this macro.

File: src/lib.rs [hunk 8]
```diff
@@ -223,16 +202,15 @@ fn bind_idents_labels(
...
     if inner_value != None {
         // possible if/let
         let mut viewed_lines_mutable = viewed_lines.clone();
-        dbg!(&viewed_lines_mutable);
         let last_ident_untrimmed = viewed_lines_mutable.pop().unwrap();
         let last_ident = last_ident_untrimmed
             .trim()
             .splitn(2, ':')
             .collect::<Vec<&str>>()[0]
+            .trim()
```
Comment (dannasessha) [inline src/lib.rs:213]: Should we leave some kind of note about the space this line is addressing? Should it be an issue like the issues we are identifying with the scrubber?

**Comment by dannasessha** [comment]
Looks good!

**Pull request merged**